source/geod24/concurrency.d

```diff
@@ -925,35 +925,45 @@ protected:
     static class InfoFiber : Fiber
     {
         ThreadInfo info;
+        FiberCondition cond;
 
         this(void delegate() op, size_t sz = 512 * 1024) nothrow
         {
+            cond = null;
             super(op, sz);
         }
     }
 
     protected class FiberCondition
     {
+        MonoTime limit = MonoTime.init;
+
+        private void registerToInfoFiber() nothrow
```

Comment:
I wonder if you can move the registering to the `FiberCondition` constructor? That way we don't have to remember to call `registerToInfoFiber()` in all the `wait` routines. So basically we could turn `registerToInfoFiber` into a constructor and remove the calls to `registerToInfoFiber`.

Reply:
That would be problematic, if the Fiber creates a `FiberBinarySemaphore` and yields before calling `wait` for whatever reason. `FiberScheduler` will still think that it is waiting on the semaphore.